include/xtensor/xmath.hpp

```diff
@@ -2307,6 +2346,42 @@ namespace detail {
             }
         };
 
+        struct nan_min
+        {
+            template <class T, class U>
+            constexpr auto operator()(const T lhs, const U rhs) const
+                -> std::common_type_t<T, U>
+            {
+                if (math::isnan(lhs))
+                {
+                    return rhs;
+                }
+                if (math::isnan(rhs))
+                {
+                    return lhs;
+                }
+                return math::minimum<void>()(lhs, rhs);
```

Comment:
I guess we need something different here to enable SIMD, but am not sure how to do it.